fe/fe-core/src/main/java/com/starrocks/connector/ConnectorFactory.java

```diff
@@ -34,11 +34,30 @@ public class ConnectorFactory {
      * @param context - encapsulate all information needed to create a connector
      * @return a connector instance
      */
-    public static CatalogConnector createConnector(ConnectorContext context) throws StarRocksConnectorException {
+    public static CatalogConnector createConnector(ConnectorContext context, boolean isReplay)
+            throws StarRocksConnectorException {
         if (null == context || !ConnectorType.isSupport(context.getType())) {
             return null;
         }
 
+        try {
+            LazyConnector lazyConnector = new LazyConnector(context);
+            if (!isReplay) {
+                lazyConnector.initIfNeeded();
```

Comment:
I think it should be caller's duty to trigger this function instead so no flag as `isReplay` needed.